From d3d2b2b69ac04ac1d1ead30f546fb4884fb93e27 Mon Sep 17 00:00:00 2001
From: Renjith Ravindran <Renjith.RavindranKannath@amd.com>
Date: Mon, 8 Aug 2022 22:26:13 +0000
Subject: [PATCH] Improve compilation without git repo and remove compiler rt
 linkage for host and correction in CMake target path variable

---
 bin/hipcc.pl               |  3 ++-
 hipamd/CMakeLists.txt      | 12 +++++++++---
 hipamd/hip-config.cmake.in |  1 -
 3 files changed, 11 insertions(+), 5 deletions(-)

diff --git a/bin/hipcc.pl b/bin/hipcc.pl
index 1ef3a90..995abe5 100755
--- a/bin/hipcc.pl
+++ b/bin/hipcc.pl
@@ -605,7 +605,8 @@ if($HIP_PLATFORM eq "amd"){
             $targetsStr = $ENV{HCC_AMDGPU_TARGET};
         } elsif (not $isWindows) {
             # Else try using rocm_agent_enumerator
-            $ROCM_AGENT_ENUM = "${ROCM_PATH}/bin/rocm_agent_enumerator";
+            $ROCMINFO_PATH = $ENV{'ROCMINFO_PATH'} // $ROCM_PATH;
+            $ROCM_AGENT_ENUM = "${ROCMINFO_PATH}/bin/rocm_agent_enumerator";
             $targetsStr = `${ROCM_AGENT_ENUM} -t GPU`;
             $targetsStr =~ s/\n/,/g;
         }
diff --git a/hipamd/CMakeLists.txt b/hipamd/CMakeLists.txt
index 9591924..4f50c2a 100755
--- a/hipamd/CMakeLists.txt
+++ b/hipamd/CMakeLists.txt
@@ -91,7 +91,13 @@ string(REPLACE "-" ";" VERSION_LIST ${HIP_VERSION_PATCH_GITHASH})
 list(GET VERSION_LIST 0 HIP_VERSION_PATCH)
 set(HIP_VERSION_GITDATE 0)
 
-find_package(Git)
+if (IS_DIRECTORY "${PROJECT_SOURCE_DIR}/.git")
+  find_package(Git)
+endif()
+set(HIP_VERSION_GITDATE "0")
+set(HIP_VERSION_GITHASH "0")
+set(HIP_VERSION_BUILD_ID 0)
+set(HIP_VERSION_BUILD_NAME "")
 
 # FIXME: Two different version strings used.
 # Below we use UNIX commands, not compatible with Windows.
@@ -183,7 +189,7 @@ set (HIP_LIB_VERSION_MINOR ${HIP_VERSION_MINOR})
 if (${ROCM_PATCH_VERSION} )
    set (HIP_LIB_VERSION_PATCH ${ROCM_PATCH_VERSION})
 else ()
-   set (HIP_LIB_VERSION_PATCH ${HIP_VERSION_PATCH}-${HIP_VERSION_GITHASH})
+   set (HIP_LIB_VERSION_PATCH "0")
 endif ()
 set (HIP_LIB_VERSION_STRING "${HIP_LIB_VERSION_MAJOR}.${HIP_LIB_VERSION_MINOR}.${HIP_LIB_VERSION_PATCH}")
 if (DEFINED ENV{ROCM_RPATH})
@@ -395,7 +401,7 @@ if(NOT ${INSTALL_SOURCE} EQUAL 0)
     if(WIN32)
       install(DIRECTORY ${HIP_COMMON_DIR}/cmake DESTINATION .)
     else()
-      install(DIRECTORY ${HIP_COMMON_DIR}/cmake/ DESTINATION CONFIG_PACKAGE_INSTALL_DIR)
+      install(DIRECTORY ${HIP_COMMON_DIR}/cmake/ DESTINATION ${CONFIG_PACKAGE_INSTALL_DIR})
     endif()
 endif()
 
diff --git a/hipamd/hip-config.cmake.in b/hipamd/hip-config.cmake.in
index ba3e75c..02dd586 100755
--- a/hipamd/hip-config.cmake.in
+++ b/hipamd/hip-config.cmake.in
@@ -290,7 +290,6 @@ if(HIP_COMPILER STREQUAL "clang")
     if(CLANGRT_BUILTINS-NOTFOUND)
       message(FATAL_ERROR "clangrt builtins lib not found")
     else()
-      set_property(TARGET hip::host APPEND PROPERTY INTERFACE_LINK_LIBRARIES "${CLANGRT_BUILTINS}")
       set_property(TARGET hip::device APPEND PROPERTY INTERFACE_LINK_LIBRARIES "${CLANGRT_BUILTINS}")
     endif()
   endif()
-- 
2.25.1

